File: backend/app/Models/SaveHotel.php
<?php 

namespace App\Models; 

use Illuminate\Database\Eloquent\Model; 
use Illuminate\Database\Eloquent\Factories\HasFactory; 
use App\Models\User; 
use App\Models\Hotel; 

class SaveHotel extends Model{
    use HasFactory; 

    protected $table = 'save_hotel'; 

    protected $primaryKey = 'id_savehotel';

    protected $keyType = 'int';

    public $timestamps = false;

    protected $fillable = [
        'id_user',
        'id_hotel',
        'is_saved',
    ];

    public function user(){
        return $this->belongsTo(User::class, 'id_user', 'id_user'); 
    }
    public function hotel(){
        return $this->belongsTo(Hotel::class, 'id_hotel', 'id_hotel'); 
    }
}

File: frontend/lib/entity/hotel/save_hotel.dart
import 'hotel.dart';

class SaveHotel{
  final int idSaveHotel;
  final int idUser;
  final int idHotel;
  final bool isSaved;
  final Hotel? hotel;

  SaveHotel({
    required this.idSaveHotel,
    required this.idUser,
    required this.idHotel,
    required this.isSaved,
    this.hotel,
  });

  factory SaveHotel.fromJson(Map<String, dynamic> json) {
    return SaveHotel(
      idSaveHotel: json['id_savehotel'],
      idUser: json['id_user'],
      idHotel: json['id_hotel'],
      isSaved: json['is_saved'] == 1 || json['is_saved'] == true || json['is_saved'] == '1',
      hotel: json['hotel'] != null ? Hotel.fromJson(json['hotel']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_savehotel': idSaveHotel,
      'id_user': idUser,
      'id_hotel': idHotel,
      'is_saved': isSaved ? 1 : 0,
      'hotel': hotel,
    };
  }
}
